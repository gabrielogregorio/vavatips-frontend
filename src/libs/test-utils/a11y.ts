import { configureAxe } from 'jest-axe';

export const a11yValidations = configureAxe({
  rules: {},
});
