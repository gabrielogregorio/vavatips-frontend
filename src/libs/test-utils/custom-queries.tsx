import { queryHelpers, buildQueries, Matcher, MatcherOptions } from '@testing-library/react';

const queryAllByDataCy = (container: HTMLElement, id: Matcher, options?: MatcherOptions | undefined) =>
  queryHelpers.queryAllByAttribute('data-cy', container, id, options);

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const getMultipleError = (c, dataCyValue) => `Found multiple elements with the data-cy attribute of: ${dataCyValue}`;
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const getMissingError = (c, dataCyValue) => `Unable to find an element with the data-cy attribute of: ${dataCyValue}`;

const [queryByDataCy, getAllByDataCy, getByDataCy, findAllByDataCy, findByDataCy] = buildQueries(
  queryAllByDataCy,
  getMultipleError,
  getMissingError,
);

export { queryByDataCy, queryAllByDataCy, getByDataCy, getAllByDataCy, findAllByDataCy, findByDataCy };
