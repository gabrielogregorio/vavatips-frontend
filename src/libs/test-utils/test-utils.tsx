import { render } from '@testing-library/react';
import { queries, within, RenderOptions } from '@testing-library/react';
import * as customQueries from './custom-queries';
import { ReactElement } from 'react';
import React from 'react';
import userEvent from '@testing-library/user-event';
import { DirectOptions } from '@testing-library/user-event/dist/cjs/setup/directApi.js';
import { UserEventApi } from '@testing-library/user-event/dist/cjs/setup/setup.js';

const allQueries = {
  ...queries,
  ...customQueries,
};

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    // <ThemeProvider theme="light">
    // <TranslationProvider messages={defaultStrings}>
    <div>{children}</div>
    // </TranslationProvider>
    // </ThemeProvider>
  );
};

const customScreen = within(document.body, allQueries);
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const customWithin = (element: ReactElement) => within(element, allQueries);
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) =>
  render(ui, { wrapper: AllTheProviders, queries: allQueries, ...options });

const click = async (element: Element, options?: DirectOptions) => userEvent.click(element, options);
const clear = async (element: Element) => userEvent.clear(element);
const type = async (element: Element, text: string, options?: DirectOptions & Parameters<UserEventApi['type']>[2]) =>
  userEvent.type(element, text, options);

// export * from '@testing-library/react';
export { customScreen as screen, customWithin as within, customRender as render, click, type, clear };
