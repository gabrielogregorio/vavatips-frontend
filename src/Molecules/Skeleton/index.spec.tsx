import { Skeleton } from '.';
import { render } from '../../libs/test-utils/test-utils';

describe('Skeleton', () => {
  it('should render a default variant', async () => {
    const { container } = render(<Skeleton />);

    expect(container).toMatchSnapshot();
  });
});
