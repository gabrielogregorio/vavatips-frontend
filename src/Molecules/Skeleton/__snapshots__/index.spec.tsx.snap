// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should render a default variant 1`] = `
<div>
  <div>
    <div
      class="bg-content-bg-disabled animate-pulseIn300 w-[220px] h-[184px]"
    />
  </div>
</div>
`;
